omod/src/main/java/org/openmrs/module/reporting/web/datasets/ManageDatasetsController.java

```diff
@@ -41,7 +46,20 @@ public String editDataSet(
 	
 		DataSetDefinitionService service = Context.getService(DataSetDefinitionService.class);
 		DataSetDefinition dsd = service.getDefinition(uuid, type);
+		List<ReportDefinition> reportDefinitions = Context.getService(ReportDefinitionService.class).getAllDefinitions(false);
+		List<ReportDefinition> affectedReportDefs = new ArrayList<ReportDefinition>();
+		for (ReportDefinition reportDef : reportDefinitions) {
+			List<Mapped<? extends DataSetDefinition>> dsds = new ArrayList<Mapped<? extends DataSetDefinition>>(reportDef.getDataSetDefinitions().values());
+			if (!dsds.isEmpty()) {
+				for (Mapped<? extends DataSetDefinition> dataSet : dsds) {
+					if (uuid.equals(dataSet.getUuidOfMappedOpenmrsObject())) {
+						affectedReportDefs.add(reportDef);
+					}
+				}
```

Comment:
If we are actually loading all of the ReportDefinitions and DSDs here, then we don't need to compare uuids, we should just do an equality check of the dataset.  By comparing UUIDs, what we meant was that, instead of loading all ReportDefinition objects, we would check for the presence of a given uuid in the serialized_data column of the serialized_object table.  You should be able to use the serializeddefinitionservice, get all serialized definitions in raw form, do a String comparison of the serializedData field, and then return matches from this.  The downside is that this might "miss" definitions that are not stored in the serialized object table.  The upside is that it will find more than just report definitions (eg. if a data set refers to another data set, or an indicator refers to a data set, etc).  And it will be much faster to run and load.

Reply:
@mseaton Thanks for reviewing my commit. Am sort of a newbie to OpenMRS and happy to contribute to the Reporting module. I took time looking for such tables: 
1: Report
2:Report defs 
3:dataSets etc...
Little did I know that they are persisted in "serialized_objects" table. Meaning I need to make a method in the "serializeddefinitionservice" that filters and retrieves reportDefs with dataSet of given UUID. I'm I on the right track with my raw mindset on this?

Reply:
@samuelmale - yes, that could be a way to go.  That being said, I'm not 100% sure that the SerializedObjectDAO has a method to retrieve objects whose serialized data matches a particular expression, so this might also require an update to the serialization.xstream module to support that.  This could be an optimization for later, which we don't need to do now.  I guess I'd stick with your current plan as it is and we can improve from there as needed.  That being said, this might be slow, so I'd be a bit cautious about adding this function to the main data set editor page.  It might be better to start off with it as an optional feature that needs to be clicked into in order to load it.

Reply:
@mseaton Thanks for that advise given. Am sorry for the late response but I had been pretty blocked by illness for 4 days. According to the mentioned above in your comment, I think we should add a link or button that one clicks to load the affected reports, is it so? If its so am good, just gonna implement that, otherwise please guide me. Thanks